request.js

```diff
@@ -0,0 +1,42 @@
+var instagramPosts = require('./posts.js')
+var https = require('https')
+
+function fetchImages(instagramPosts, callback) {
+  var instagramImages = []
+  var counter = 0
+  var numberOfPosts = instagramPosts.length
+  instagramPosts.forEach(function(post, index){
+    var timeStart = Date.now()
+    https.get(post, function(response) {
+      var body = ''
+      response.on('data', function(chunk) {
+        body += chunk
+      })
+      response.on('end', function () {
```

Comment:
I think this lab does a great job of getting users to feel the pain of callback-based async before introducing them to something like promises.